samcli/commands/local/lib/local_lambda.py

```diff
@@ -86,6 +88,33 @@ def is_debugging(self):
         """
         return bool(self.debug_context)
 
+    def _sanitize_inlinecode(self, inlinecode):
```

Comment:
What if we did the InlineCode sanitizing while reading the template instead? We could create a provider that is responsible for understanding InlineCode and produces the file (lambda_handler) and updated template. Then everything will magically work.

This could help in the future if we wanted to load or 'pre-build' InlineCode during the `sam local [start-lambda|start-api]` commands.

Reply:
makes completely sense. I just tried to hack around a quick way to get the job done but creating an object holding the code is definitively more elegant.. let me wrap around when i find few free hours to dedicate to it

Reply:
I've just rewrote  the skel. Is this what you actually would like to see?
The CodeProvider is a context manager that should be used before `self.local_runtime.invoke`, in `samcli/commands/local/lib/local_lambda.py`